common/taskvalidator/validateworkflow.go

```diff
@@ -24,62 +24,93 @@
 package taskvalidator
 
 import (
+	"context"
+
+	"go.uber.org/zap"
+
 	"github.com/uber/cadence/common/cache"
 	"github.com/uber/cadence/common/log"
 	"github.com/uber/cadence/common/log/tag"
 	"github.com/uber/cadence/common/metrics"
+	"github.com/uber/cadence/common/persistence"
+	"github.com/uber/cadence/common/reconciliation/invariant"
+	"github.com/uber/cadence/common/types"
 )
 
-// Checker is an interface for initiating the validation process.
 type Checker interface {
-	WorkflowCheckforValidation(workflowID string, domainID string, domainName string, runID string) error
+	WorkflowCheckforValidation(ctx context.Context, workflowID string, domainID string, domainName string, runID string) error
+}
+
+// Define the staleChecker interface with the methodsgit s.
+type staleChecker interface {
+	CheckAge(response *persistence.GetWorkflowExecutionResponse) (bool, error)
 }
 
-// checkerImpl is the implementation of the Checker interface.
 type checkerImpl struct {
 	logger        log.Logger
 	metricsClient metrics.Client
 	dc            cache.DomainCache
+	pr            persistence.Retryer
+	staleCheck    staleChecker
 }
 
-// NewWfChecker creates a new instance of Checker.
-func NewWfChecker(logger log.Logger, metrics metrics.Client, domainCache cache.DomainCache) Checker {
+func NewWfChecker(logger log.Logger, metrics metrics.Client, domainCache cache.DomainCache, pr persistence.Retryer) Checker {
+	zapLogger, _ := zap.NewProduction()
```

Comment:
we shouldn't create side loggers like this. can `NewStaleWorkflow` accept `log.Logger` instead?

Reply:
it wasn't able to but maybe with the new stalechecker it will. I will check.

Reply:
I see you changed constructor to expect `*zap.Logger`. that doesn't address my concern. we should keep it consistent via `log.Logger`. let's look into changing the stale checker to accept `log.Logger` not the other way around